src/components/TimeRangePicker/TimeRangePicker.js

```diff
@@ -0,0 +1,80 @@
+import PropTypes from "prop-types";
+import { useEffect, useState } from "react";
+import { FiClock } from "react-icons/fi";
+import { MdOutlineKeyboardArrowDown } from "react-icons/md";
+import clsx from "clsx";
+import { TimeRangePickerBody } from "./TimeRangePickerBody";
+import "./index.css";
+import { defaultRange } from "./rangeOptions";
+import { convertRangeValue, createDisplayValue, storage } from "./helpers";
+
+export const TimeRangePicker = ({ onChange }) => {
+  const [isPickerOpen, setIsPickerOpen] = useState(false);
+  const [currentRange, setCurrentRange] = useState(
+    storage.getItem("currentRange") || defaultRange
+  );
+  const [rangeDisplayValue, setRangeDisplayValue] = useState(
+    createDisplayValue(currentRange)
+  );
+
+  const changeRangeValue = (range) => {
+    const { from, to } = range;
+    setCurrentRange({ from, to });
+    onChange(
+      convertRangeValue(from, "jsDate"),
+      convertRangeValue(to, "jsDate")
+    );
+  };
+
+  useEffect(() => {
```

Comment:
I think this useEffect is not needed 
```suggestion
  useEffect(() => {
```